arctic/store/_pandas_ndarray_store.py

```diff
@@ -346,3 +355,218 @@ def read(self, arctic_lib, version, symbol, **kwargs):
+
+    def can_write(self, version, symbol, data, **kwargs):
+        if isinstance(data, (DataFrame, Series)):
+            if 'date' in data.index.names and 'chunk_size' in version and version['chunk_size'] in ('D', 'M', 'Y'):
+                if isinstance(data, DataFrame) and np.any(data.dtypes.values == 'object'):
+                    return self.can_convert_to_records_without_objects(data, symbol, PandasDataFrameStore().to_records)
+                elif isinstance(data, Series) and (data.dtype == np.object_ or data.index.dtype == np.object_):
+                    return self.can_convert_to_records_without_objects(data, symbol, PandasSeriesStore().to_records)
+                return True
+        return False
+
+    def can_update(self, version, symbol, data, **kwargs):
+        if self.can_read(version, symbol):
+            if isinstance(data, DataFrame) and version['pandas_type'] == 'df' \
+                    or isinstance(data, Series) and version['pandas_type'] == 'series':
+                return self.can_write(version, symbol, data, **kwargs)
+        return False
+
+    def get_date_chunk(self, date, chunk_size):
+        '''
+        format date appropriately for the chunk size
+        
+        returns
+        -------
+        Formatted date string
+        '''
+        if chunk_size == 'Y':
+            return date.strftime('%Y')
+        elif chunk_size == 'M':
+            return date.strftime('%Y-%m')
+        elif chunk_size == 'D':
+            return date.strftime('%Y-%m-%d')
+
+
+    def get_range(self, df):
+        """
+        get minx/max dates in the index of the dataframe
+        
+        returns
+        -------
+        A tuple (start date, end date)
+        """
+        dates = df.index.get_level_values('date')
+        start = dates.min()
+        end = dates.max()
+        if isinstance(start, Timestamp):
+            start = start.to_pydatetime()
+        if isinstance(end, Timestamp):
+            end = end.to_pydatetime()
+        return start, end
+
+    def to_records(self, df, chunk_size):
+        """
+        chunks the dataframe/series by dates
+
+        returns
+        -------
+        A list of tuples - (start date, end date, dataframe/series)
+        """
+        dates = [pd.to_datetime(d) for d in df.index.get_level_values('date').drop_duplicates()]
+        key_array = [self.get_date_chunk(d, chunk_size) for d in dates]
+
+        for date in set(key_array):
+            if df.index.nlevels > 1:
+                '''
+                can't slice with partial date on multi-index. Support coming in 
+                pandas 0.18.1
+                '''
+                ret = df.xs(slice(date, date), level='date', drop_level=False)
+            else:
+                ret = df[date : date]
+            start, end = self.get_range(ret)
+            yield start, end, ret
+
+    def write(self, arctic_lib, version, symbol, item, previous_version, chunk_size=None):
```

Comment:
yes, you cannot change the chunk size, or should not be able to. Sounds like some new unit tests are in order.